File: java/src/WikiPageResponder.java
import java.util.List;

import sun.reflect.generics.reflectiveObjects.NotImplementedException;


class WikiPageResponder extends ResultResponder {
	public Response make_response(Request request, RequestContext context) {
		List<WikiPage> all_pages = new DepthFirstTraverser(context.root_page())
				.traverse();
		for (WikiPage p : all_pages) {
			if (p.uri().equals(request.uri()))
				return new Response(p, "200");
		}
		return new Response(new WikiPage("404 page"), "404");
	}

	@Override
	protected String title() {
		throw new NotImplementedException();
	}

	@Override
	protected boolean traverse(WikiPage page) {
		throw new NotImplementedException();
	}
}

File: csharp/WikiPageResponder.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;

namespace WikiSearch
{
	public class WikiPageResponder : ResultResponder
	{
		public override Response MakeResponse(Request request, RequestContext context)
		{
			List<WikiPage> allPages = new DepthFirstTraverser(context.RootPage).Traverse();
			foreach (WikiPage p in allPages)
				if (p.Uri == request.Uri)
					return new Response(p, "200");
			return new Response(new WikiPage("404 page"), "404");
		}

		public override string Title()
		{
			throw new NotImplementedException();
		}

		public override bool Traverse(WikiPage page)
		{
			throw new NotImplementedException();
		}
	}
}

